src/smart-auth/index.ts

```diff
@@ -119,14 +136,17 @@ function routeCase(value: string): string {
 }
 
 const oauthPlugin: FastifyPluginCallback<SmartAuthProvider> = function (http, options, next) {
-  const { name, auth, client, scope: defaultScope, redirect } = options
+  const { name, client } = options;
+  supports(options, "authorization_code");
```

Comment:
do we want to assert this? or `if (!supports(options, 'authorization_code')) { throw new Error(...); }`

Reply:
Yeah not a bad suggestion esp. if consumed outside of TS